peachjam/views/documents.py

```diff
@@ -9,6 +11,48 @@
 from peachjam.registry import registry
 
 
+class RedirectResolver:
+
+    RESOLVER_MAPPINGS = {
+        "africanlii": {"country_code": "aa", "domain": "agp.africanlii.org"},
+        "ghalii": {"country_code": "gh", "domain": "new.ghalii.org"},
+        "lawlibrary": {"country_code": "za", "domain": "lawlibrary.org.za"},
+        "lesotholii": {"country_code": "ls", "domain": "new.lesotholii.org"},
+        "malawilii": {"country_code": "mw", "domain": "malawilii.org"},
+        "namiblii": {"country_code": "na", "domain": "namiblii.org"},
+        "senlii": {"country_code": "sn", "domain": "senlii.org"},
+        "seylii": {"country_code": "sc", "domain": "new.seylii.org"},
+        "sierralii": {"country_code": "sl", "domain": "new.sierralii.org"},
+        "tanzlii": {"country_code": "tz", "domain": "new.tanzlii.org"},
+        "tcilii": {"country_code": "tc", "domain": "new.tcilii.org"},
+        "ulii": {"country_code": "ug", "domain": "new.ulii.org"},
+        "zambialii": {"country_code": "zm", "domain": "new.zambialii.org"},
+        "zanzibarlii": {"place_code": "tz-znz", "domain": "new.zanzibarlii.org"},
+        "zimlii": {"country_code": "zw", "domain": "zimlii.org"},
+    }
+
+    def __init__(self):
+        mapping_code = self.RESOLVER_MAPPINGS[settings.PEACHJAM["APP_NAME"].lower()]
+        self.current_authority = mapping_code.get("place_code") or mapping_code.get(
```

Comment:
I think it'll be simpler if the current authority is either the key in the dict, or the object from the dict. That way you can compare an authority with the current authority directly, without having to re-compare place codes.

So the code comparison is done ONLY when looking for the authority that matches an FRBR URI. Once you have that authority, you compare it with the current one.

Otherwise you're juggling codes in three places, instead of one.